components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1502,6 +1502,38 @@ public void deleteIdP(String idPName, String tenantDomain) throws IdentityProvid
         }
     }
 
+    /**
+     * Force delete an Identity Provider from a given tenant. This will remove any associations this Identity
+     * Provider has with any Service Providers in authentication steps or provisioning.
+     *
+     * @param idpName name of IDP to be deleted
+     * @param tenantDomain tenantDomain to which the IDP belongs to
+     */
+    public void forceDeleteIdp(String idpName, String tenantDomain) throws IdentityProviderManagementException {
+
+        // invoking the pre listeners
+        Collection<IdentityProviderMgtListener> listeners = IdPManagementServiceComponent.getIdpMgtListeners();
+        for (IdentityProviderMgtListener listener : listeners) {
+            if (listener.isEnable() && !listener.doPreDeleteIdP(idpName, tenantDomain)) {
```

Comment:
Shall we add some debug logs printing which listener is getting executed at least ? This will help us to identify if an issue occurs within a listener without depending on the listener implementation logs.

Reply:
+1 will add